### ISSUE TITLE
Need CNAME cloaking

### ISSUE THREAD

**Issue by Nefurtity** [DNSCrypt/dnscrypt-proxy#209]
Cloaking only applies to A or AAAA record, but not CNAME record.
Many sites manage their host name by a CNAME record.

For example, ```ajax.googleapis.com``` has a CNAME for ```googleapis.l.google.com```, also ```play.googleapis.com``` has a same CNAME for ```googleapis.l.google.com``` too.

Supporting a CNAME cloaking rule can reduce much work for those host names has a same CNAME.

eg. 
```
ajax.googleapis.com 127.0.0.1
play.googleapis.com 127.0.0.1
```

Can replace by only one line.
```
googleapis.l.google.com 127.0.0.1
```

**Comment by iWARR** [comment]
Looks very interesting.

I want to understand the way how it MAY work.
In theory (by your example), instead of having a blocking rules in the `'blacklist.txt'` for:
```
ajax.googleapis.com
play.googleapis.com
```

I can try to use cloacking rule:
`
googleapis.l.google.com 127.0.0.1` (or, even better `0.0.0.0` w/o lookup)

Right? Will blocking work for 100% ?

---------------

**Note:**

I don't know, if `ajax.googleapis.com` just an example here, but in the "real life" blocking this domain is not a good idea. This will break the main functionality of the webpage (I know it for 100%). The good idea is blocking "cookies" only for this domain to avoid spying on you. 

Some of the spying domains you can easely block completely, some of them - not (only partially: cookies, js, etc).

Proof: https://better.fyi/trackers/ajax.googleapis.com/

**Comment by Nefurtity** [comment]
Actually I don't want to block these domains, I just want to redirect these domains to a local reverse proxy, just like sni proxy or something like.
Because these domains are blocked by a firewall, I need a proxy to access these blocked sites.

**Comment by iWARR** [comment]
Oh... I see, this is another story.
dnscrypt-proxy can be very flexible, you know :)

If any, be careful, local addresses are under attack now. Hope, you have no incoming in this case.
For example, Chrome/uTorrent: https://bugs.chromium.org/p/project-zero/issues/detail?id=1524

**Comment by jedisct1** [comment]
Being able to redirect `*.googleapis.l.google.com` would indeed be very useful.

If you can contribute this feature, please do!
